python/packages/autogen-agentchat/src/autogen_agentchat/teams/_group_chat/_graph/_digraph_group_chat.py

```diff
@@ -37,12 +37,42 @@ class DiGraphEdge(BaseModel):
     """
 
     target: str  # Target node name
-    condition: str | None = None  # Optional execution condition (trigger-based)
+    condition: Union[str, Callable[[BaseChatMessage], bool], None] = None
     """(Experimental) Condition to execute this edge.
-    If None, the edge is unconditional. If a string, the edge is conditional on the presence of that string in the last agent chat message.
-    NOTE: This is an experimental feature WILL change in the future releases to allow for better spcification of branching conditions
-    similar to the `TerminationCondition` class.
+    If None, the edge is unconditional.
+    If a string, the edge is conditional on the presence of that string in the last agent chat message.
```

Comment:
Same as previous comment. Remove the string as an option. Only callable can be used as condition.